### ISSUE TITLE
Add separators between generated entries to ease being manipulating changelog files

### ISSUE THREAD

**Issue by josephperrott** [angular/dev-infra#198]
Add a separator between changelog entries in CHANGELOG.md files to allow us to easily do things like archiving or filtering when updating changelogs.

Proposal:
```md
<a name="1.2.3"></a>
# 1.2.3 (2000-01-01)
### ng-dev
| Commit | Type | Description |
| -- | -- | -- |
| [abc123](https://github.com/angular/dev-infra/commit/abc123) | feat | **release:** Add changlog separators |

<!-- CHANGELOG SPLIT MARKER -->

<a name="1.2.2"></a>
# 1.2.2 (2000-01-01)
```

**Comment by devversion** [comment]
Could we use the anchor instead? that seems like it would be sufficient already?

**Comment by josephperrott** [comment]
We can, the only concern I see with using the anchor is that splitting would be more complicated since we would have to do a regex match for the split.  But we could definitely code around that if needed.  We could also make it easier to identify with a marker in the anchor tag `<a versionMarker name="1.2.3"></a>` and we could match on "`\<a versionMarker `"

```ts
/** The changelog contents. */
let changelogContents: string;
/** The marker used as a separator between entries. */
const versionMarker = `<a versionMarker `;
/** The separated entry list. */
const changelogEntries: string[] = changelogContents
  // Use the versionMarker as the separator for .split().
  .split(versionMarker)
  // If the `split()` method finds the separator at the beginning or end of a string, it includes
  // an empty string at the respective locaiton, so we filter to remove all of these potential
  // empty strings.
  .filter(entry => entry.length !== 0)
  // Because .split() removes the separator that was discovered, we need to put in back in place.
  .map; (entry => `${versionMarker}${entry}`);
```

**Comment by devversion** [comment]
yeah, maybe the following could also work:

```js
changelog.split(/<a name="[^"]+"><\/a>/g)
```

I see how an actual marker comment is more future proof though.

**Comment by josephperrott** [comment]
I believe that with the separator you provide, we would love the version that was split when reconstructing the changelog back together.

**Comment by devversion** [comment]
yeah, you're right. Also using the anchor is not as future-proof. e.g. if we change the attribute from name to id.

**Comment by josephperrott** [comment]
Closed by #212
